python_modules/dagster-graphql/dagster_graphql/schema/auto_materialize_asset_evaluations.py

```diff
@@ -65,23 +70,43 @@ class Meta:
         )
 
 
-def create_graphene_auto_materialize_condition(condition: AutoMaterializeCondition):
-    if isinstance(condition, FreshnessAutoMaterializeCondition):
-        return GrapheneFreshnessAutoMaterializeCondition(decisionType=condition.decision_type)
-    elif isinstance(condition, DownstreamFreshnessAutoMaterializeCondition):
-        return GrapheneDownstreamFreshnessAutoMaterializeCondition(
-            decisionType=condition.decision_type
-        )
-    elif isinstance(condition, ParentMaterializedAutoMaterializeCondition):
-        return GrapheneParentMaterializedAutoMaterializeCondition(
-            decisionType=condition.decision_type
+DAGSTER_NAMEDTUPLE_TO_GRAPHENE_MAP = {
+    FreshnessAutoMaterializeCondition: GrapheneFreshnessAutoMaterializeCondition,
+    DownstreamFreshnessAutoMaterializeCondition: GrapheneDownstreamFreshnessAutoMaterializeCondition,
+    ParentMaterializedAutoMaterializeCondition: GrapheneParentMaterializedAutoMaterializeCondition,
+    MissingAutoMaterializeCondition: GrapheneMissingAutoMaterializeCondition,
+    ParentOutdatedAutoMaterializeCondition: GrapheneParentOutdatedAutoMaterializeCondition,
+}
+
+
+def create_graphene_auto_materialize_condition(
+    condition: Union[AutoMaterializeCondition, Tuple[AutoMaterializeCondition, str]],
+    partitions_def: Optional[PartitionsDefinition],
+):
+    if partitions_def is not None:
+        if not len(condition) == 2 and isinstance(condition[1], str):
```

Comment:
In the event that we add a new field to one of these `AutoMaterializeCondition`s whose type is `str`, then this check will incorrectly identify that as a the second case.

Personally, I prefer the:

`isinstance(condition, AutoMaterializeCondition.__args__)`

thing that you had previously, as it's guaranteed to stay in sync with the underlying AutoMaterializeCondition type. Any reservations with that approach?